R/exec.R

```diff
@@ -0,0 +1,48 @@
+#' Execute a function
+#'
+#' @description
+#' This function constructs and evaluates a call to `f`.
+#' It was two primary uses:
+#'
+#' * To call a function with arguments stored in a list (if the function
+#'   doesn't support [tidy-dots])
+#'
+#' * To call every function stored in a list (in conjunction with `map()`/
+#'   [lapply()])
+#'
+#' @param f A function, or function name as a string.
+#' @param ... Arguments to function.
+#'
+#'   These dots support [tidy-dots] features.
+#' @param  .env Environment in which to evaluate the call. This will be
+#'   most useful if `f` is a string, or the function as side-effects.
+#' @export
+#' @examples
+#' args <- list(x = c(1:10, 100, NA), na.rm = TRUE)
+#' exec("mean", !!!args)
+#' exec("mean", !!!args, trim = 0.2)
+#'
+#' fs <- list(a = function() "a", b = function() "b")
+#' lapply(fs, exec)
+#'
+#' # Compare to do.call it will not automatically inline expressions
+#' # into the evaluated call.
+#' x <- 10
+#' args <- exprs(x1 = x + 1, x2 = x * 2)
+#' exec(list, !!!args)
+#' do.call(list, args)
```

Comment:
Does it need an example to show how supplying a string creates better backtraces / recorded calls?

And then show how to construct and evaluate a call manually when it's important to mask the arguments as well. Maybe taking `lm()` as example:

```r
f <- disp ~ cyl

# Horrible call
exec("lm", f, data = mtcars)

# Nice call
eval(expr(lm(!!f, data)), list(data = mtcars))
```